bionic/flow.py

```diff
@@ -603,8 +603,8 @@ def merge(self, flow, keep='error', allow_name_match=False):
         # For both states, check that each jointly-defined name group is kept
         # or discarded as a whole.
         for state_name, state in [
-                    ('self', old_state),
```

Comment:
@jqmp I reverted the state_names to the originals, since ultimately this is just an api level change.